comments/loading_perf.md

```diff
@@ -0,0 +1,114 @@
+* 420ms HTTP GET blobstore tiles (~64KB in parallel)
+* ~70ms JS `hasRAGData`
+
+Time to get tiles from the blobstore can vary wildly - have seen up to 2000ms to complete - but they do load incrementally into the canvas under labels and the page is responsive by then.
+
+
+## Proposals in detail
+
+### No reshape
+
+A majority of the time spent due to having a larger map model is not getting the model to the browser, but making changes to it once it had loaded. The reshape code has been designed so that the frontend code can grown new expectations about the shape and required ontology for maps, without imposing the cost of revisiting all older maps to update for all migrations. For large and frequently updated maps, we should handle these ongoing expectation changes at the project level instead.
+
+Skipping the reshape step entirely can be a flag that exists on the model (or in local storage for testing), and will be a better option than optimising the existing reshape steps which perform fine for the small historical maps they were designed for.
+
+There are several updates required for any model wishing to skip the current reshape steps:
+
+* The introduction view needs to be in the model rather than coming from the 'welcome-splash' map settings.
+* All categories need to be derived from `gist:Caterory` and `gist:isCategorizedBy` predicate in the ontology.
+* An annotation category needs to exist in the ontology.
+* The map must use lenses not layers.
+* New form geo predicates must be used.
+
+There is also some convenience linking added to lenses which may need to be replaced with a cache instead.
+
+
+### API without settings.json
+
+The page load currently relies on fetching an additional api file before loading the map model, which contains information that could instead be included in the html page template and avoid an additional network roundtrip.
+
+
+### Base template with more details from model
+
+Tiles cannot start loading until nearly all other work has been done at present, because the paths for the tiles are in the model and openlayers will only start loading after all the dependent providers and components are ready. If some values, such as what the lenses are, and what tilesets they use, are extracted from the model and included in the base template, it would be possible to serve the dual purpose of having a base page that has some useful content where the browser does not support the full range of features needed to interact with the map, and an opportunity to start loading tiles into cache in parallel with model loading.
+
+
+### Server-side pre-baked model
+
+Loading the model has a large amount of wait time currently, as the server has to do work to load the json from the database and gzip compress it before sending it onto the wire. If the model is stored in a compressed and ready to send form, the first load time will be closer to the time it takes for the client to load the file over the network.
```

Comment:
So pre-baked = compressed? Presumably we can't store this in a database any more (which is doing its own compression but not in a form that helps us), so we'd have to think about alternative storage mechanisms?

Reply:
In the wire format basically, I've reworded to try to be clearer.